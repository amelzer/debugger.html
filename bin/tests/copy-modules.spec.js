const {transformSingleFile} = require("../copy-modules")
const path = require("path")

function transform(filename) {
  return transformSingleFile(path.join(__dirname,`./fixtures/${filename}.js`))
}

describe("Copy modules", () => {
  it("transforms", () => {
    const out = transform("transforms")
    expect(out).toMatchSnapshot()
  });

  it("imports", () => {
    const out = transform("imports")
    expect(out).toMatchSnapshot()
  });

  it("single telemetry", () => {
    const out = transform("single-telemetry-import")
    expect(out).toMatchSnapshot()
  });

  it("default telemtry", () => {
    const out = transform("default-telemetry-import")
    expect(out).toMatchSnapshot()
  });
});
