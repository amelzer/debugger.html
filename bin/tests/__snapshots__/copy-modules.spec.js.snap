// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy modules default telemtry 1`] = `
"\\"use strict\\";

var _devtoolsModules = require(\\"devtools/client/debugger/new/dist/vendors\\").vendored[\\"devtools-modules\\"];

var _devtoolsModules2 = _interopRequireDefault(_devtoolsModules);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_devtoolsModules2.default;"
`;

exports[`Copy modules imports 1`] = `
"\\"use strict\\";

var _devtoolsModules = require(\\"devtools/client/debugger/new/dist/vendors\\").vendored[\\"devtools-modules\\"];

var _telemetry = require(\\"devtools/client/shared/telemetry\\");

var _telemetry2 = _interopRequireDefault(_telemetry);

var _reselect = require(\\"devtools/client/debugger/new/dist/vendors\\").vendored[\\"reselect\\"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_devtoolsModules.KeyShortcuts;
_reselect.createSelector;
_telemetry2.default;"
`;

exports[`Copy modules single telemetry 1`] = `
"\\"use strict\\";

var _telemetry = require(\\"devtools/client/shared/telemetry\\");

var _telemetry2 = _interopRequireDefault(_telemetry);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_telemetry2.default;"
`;

exports[`Copy modules transforms 1`] = `
"\\"use strict\\";

const obj = {};

function simple(str) {
  const {
    a,
    b
  } = obj;
  console.log(\\"yo\\");
}

function render() {
  return React.createElement(\\"div\\", null, \\"JSX\\");
}

class klass {
  constructor() {
    this.bar = \\"okay\\";

    this.foo = () => \\"yep\\";
  }

}"
`;
